src/metrics/metrics.go

```diff
@@ -272,6 +220,64 @@ func populateCustomMetrics(instanceEntity *integration.Entity, connection *conne
 	}
 }
 
+// metricsFromCustomQueryRow obtains a map of metrics from a row resulting from a custom query.
+// A particular metric can be configured either with:
+// - Specific columns in the query: metric_name, metric_type, metric_value
+// - The corresponding `query.Name` and `query.Type`
+// When both are defined, the query columns have precedence. Besides, if type is not defined it is automatically deteced.
+// The rest of the query columns are also taken as metrics/attributes (detecting their types automatically).
+// Besides, if `query.Prefix` is defined, all metric and attribute names will include the corresponding prefix.
+func metricsFromCustomQueryRow(row []string, columns []string, query customQuery) (map[string]customQueryMetricValue, error) {
+	metrics := map[string]customQueryMetricValue{}
+
+	var metricValue string
+	metricType := query.Type
+	metricName := query.Name
+
+	for i, columnName := range columns {
+		switch columnName {
+		case "metric_name":
+			metricName = row[i]
+		case "metric_type":
+			metricType = row[i]
+		case "metric_value":
+			metricValue = row[i]
+		default:
+			name := query.Prefix + columnName
+			value := row[i]
+			metrics[name] = customQueryMetricValue{value: value, sourceType: detectMetricType(value)}
+		}
+	}
+
+	if metricValue == "" {
+		if metricName != "" {
```

Comment:
I am confused: Why `errMissingMetricNameCustomQuery` if `metricName != ""`?

Reply:
shouldn't it be errMissingMetricValueCustomQuery?

Reply:
Good catch! I swapped the errors, thanks for pointing out!

Reply:
I've just modified it